diff -urN taglib/taglib/toolkit/tiostream.cpp.orig taglib/taglib/toolkit/tiostream.cpp
--- taglib/taglib/toolkit/tiostream.cpp.orig    2015-08-06 09:55:54.882605900 +0200
+++ taglib/taglib/toolkit/tiostream.cpp 2015-08-06 10:07:33.885276100 +0200
@@ -32,6 +32,7 @@
 # include "tstring.h"
 # include "tdebug.h"
 # include <windows.h>
+# include <winapifamily.h>

 namespace
 {
@@ -40,8 +41,12 @@

   bool supportsUnicode()
   {
+#if (WINAPI_FAMILY == WINAPI_FAMILY_PC_APP || WINAPI_FAMILY == WINAPI_FAMILY_PHONE_APP)
+    return true;
+#else
     const FARPROC p = GetProcAddress(GetModuleHandleA("kernel32"), "CreateFileW");
     return (p != NULL);
+#endif
   }

   // Indicates whether the system supports Unicode file names.
