--- nfs/win32/win32_compat.h	2017-10-09 15:24:23.075684400 +0200
+++ nfs/win32/win32_compat.h.msvc81	2017-10-09 15:26:16.416474600 +0200
@@ -26,6 +26,10 @@ THE SOFTWARE.
 #define win32_COMPAT_H_
 #define NO_IPv6 1
 
+#if defined(_WIN32_WINNT) && _WIN32_WINNT <= 0x603
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
+#endif
 #include <winsock2.h>
 #include <ws2tcpip.h>
 #include <ws2ipdef.h>
