--- mpg123/configure.ac 2014-09-28 23:27:28.000000000 +0200
+++ mpg123/configure.ac.winstore        2015-08-10 15:33:35.881564100 +0200
@@ -2325,8 +2325,8 @@
   AC_MSG_CHECKING([if WINVER and _WIN32_WINNT needs version bumps])
   if test "x$win32_winver_bump" = "xyes"; then
     AC_MSG_RESULT([yes])
-    AC_DEFINE([WINVER], [0x0501], [ WinXP and above for ipv6 ])
-    AC_DEFINE([_WIN32_WINNT], [0x0501], [ WinXP and above for ipv6 ])
+    dnl AC_DEFINE([WINVER], [0x0501], [ WinXP and above for ipv6 ])
+    dnl AC_DEFINE([_WIN32_WINNT], [0x0501], [ WinXP and above for ipv6 ])
     # getaddrinfo and freeaddrinfo are available for Win2K and above
     # Bug: MinGW.org w32api ws2tcpip.h incorrectly wants (_WIN32_WINNT >= 0x0501) for getaddrinfo and freeaddrinfo
     # MultiByteToWideChar and WideCharToMultiByte are available for Win2K and above

