--- flac-1.3.2/src/libFLAC/include/private/bitmath.h	2017-01-09 13:16:13.853339600 +0100
+++ flac-1.3.2/src/libFLAC/include/private/bitmath.h.msvc	2017-01-09 12:55:42.469023300 +0100
@@ -76,16 +76,16 @@ static inline unsigned int FLAC__clz_uin
 	FLAC__ASSERT(v > 0);
 #if defined(__INTEL_COMPILER)
 	return _bit_scan_reverse(v) ^ 31U;
-#elif defined(__GNUC__) && (__GNUC__ >= 4 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 4))
-/* This will translate either to (bsr ^ 31U), clz , ctlz, cntlz, lzcnt depending on
- * -march= setting or to a software routine in exotic machines. */
-	return __builtin_clz(v);
 #elif defined(_MSC_VER)
 	{
 		unsigned long idx;
 		_BitScanReverse(&idx, v);
 		return idx ^ 31U;
 	}
+#elif defined(__GNUC__) && (__GNUC__ >= 4 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 4))
+/* This will translate either to (bsr ^ 31U), clz , ctlz, cntlz, lzcnt depending on
+ * -march= setting or to a software routine in exotic machines. */
+	return __builtin_clz(v);
 #else
 	return FLAC__clz_soft_uint32(v);
 #endif
