--- live555/groupsock/include/NetCommon.h	2017-10-09 15:05:59.507509900 +0200
+++ live555/groupsock/include/NetCommon.h.msvc81	2017-10-09 15:28:44.624677700 +0200
@@ -25,6 +25,10 @@ along with this library; if not, write t
 /* Windows */
 #if defined(WINNT) || defined(_WINNT) || defined(__BORLANDC__) || defined(__MINGW32__) || defined(_WIN32_WCE) || defined (_MSC_VER)
 #define _MSWSOCK_
+#if defined(_WIN32_WINNT) && _WIN32_WINNT <= 0x603
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
+#endif
 #include <winsock2.h>
 #include <ws2tcpip.h>
 #endif
