diff --git a/config.mingw b/config.mingw
index 18f0518..89aca02 100755
--- a/config.mingw
+++ b/config.mingw
@@ -6,12 +6,12 @@
 CPLUSPLUS_COMPILER =   $(CXX)
 CPLUSPLUS_FLAGS =      $(COMPILE_OPTS) -D__MINGW32__ -Wall -Wno-deprecated
 OBJ =                  o
-LINK =                 $(CXX) -o
+LINK =                 $(LD) -o
 LINK_OPTS =            -L.
 CONSOLE_LINK_OPTS =    $(LINK_OPTS)
-LIBRARY_LINK =         $(LD) -o
+LIBRARY_LINK =         $(AR) -o
 LIBRARY_LINK_OPTS =    $(LINK_OPTS) -r -Bstatic
-LIB_SUFFIX =                   a
+LIB_SUFFIX =                   lib
 LIBS_FOR_CONSOLE_APPLICATION = -lws2_32
 LIBS_FOR_GUI_APPLICATION = -lws2_32
 EXE =

--- a/liveMedia/RTSPCommon.cpp
+++ b/liveMedia/RTSPCommon.cpp
@@ -25,6 +25,10 @@
 #include <ctype.h> // for "isxdigit()
 #include <time.h> // for "strftime()" and "gmtime()"

+#if defined(_MSC_VER)
+#include <stdlib.h>
+#endif
+
 static void decodeURL(char* url) {
   // Replace (in place) any %<hex><hex> sequences with the appropriate 8-bit character.
   char* cursor = url;

--- a/UsageEnvironment/UsageEnvironment.cpp
+++ b/UsageEnvironment/UsageEnvironment.cpp
@@ -19,6 +19,10 @@

 #include "UsageEnvironment.hh"

+#if defined(_MSC_VER)
+#include <stdlib.h>
+#endif
+
 Boolean UsageEnvironment::reclaim() {
   // We delete ourselves only if we have no remainining state:
   if (liveMediaPriv == NULL && groupsockPriv == NULL) {

