--- libgcrypt/cipher/cipher-ccm.c	2016-05-24 12:02:08.890090300 +0200
+++ libgcrypt/cipher/cipher-ccm.c.alloca	2016-05-24 12:01:49.856208500 +0200
@@ -44,7 +44,7 @@ do_cbc_mac (gcry_cipher_hd_t c, const un
 {
   const unsigned int blocksize = 16;
   gcry_cipher_encrypt_t enc_fn = c->spec->encrypt;
-  unsigned char tmp[blocksize];
+  unsigned char *tmp = alloca(blocksize);
   unsigned int burn = 0;
   unsigned int unused = c->u_mode.ccm.mac_unused;
   size_t nblocks;
