--- upnp_clean/upnp/inc/upnp.h	2015-04-30 14:37:26.962425889 +0200
+++ upnp/upnp/inc/upnp.h	2015-04-30 14:41:48.099528162 +0200
@@ -41,6 +41,15 @@
  * \file
  */
 
+#ifdef _WIN32
+# define WIN32
+#endif
+
+#if !defined(_WIN32_WINNT) || _WIN32_WINNT != 0x501
+#undef _WIN32_WINNT
+#define _WIN32_WINNT 0x501 /* also defined in the config file as such in config.h */
+#endif /* _WIN32_WINNT */
+
 #include "ixml.h"
 #include "upnpconfig.h"
 #include "UpnpGlobal.h"
