--- upnp/configure.ac	2017-10-09 17:13:23.694147300 +0200
+++ upnp/configure.ac.msvc81	2017-10-09 17:13:19.445751600 +0200
@@ -628,6 +628,8 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
 	#include <sys/socket.h>
 #endif
 #ifdef HAVE_WS2TCPIP_H
+	#undef WINAPI_FAMILY
+	#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 	#include <ws2tcpip.h>
 #endif
 ],[ socklen_t t = 0; return t; ])
--- upnp/upnp/inc/UpnpInet.h	2017-10-09 17:13:33.082280100 +0200
+++ upnp/upnp/inc/UpnpInet.h.msvc81	2017-10-09 17:13:29.232060100 +0200
@@ -15,6 +15,8 @@
 
 #ifdef WIN32
 	#include <stdarg.h>
+	#undef WINAPI_FAMILY
+	#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 	#include <winsock2.h>
 	#include <iphlpapi.h>
 	#include <ws2tcpip.h>
