--- harfbuzz-1.0.6/configure.ac.orig	2015-10-26 00:06:30.060254321 +0100
+++ harfbuzz-1.0.6/configure.ac	2015-10-26 00:07:17.364140612 +0100
@@ -93,14 +93,6 @@
 		;;
 	esac
 
-	case "$host" in
-		arm-*-*)
-			if test "x$ac_cv_alignof_struct_char__" != x1; then
-				# Request byte alignment
-				CXXFLAGS="$CXXFLAGS -mstructure-size-boundary=8"
-			fi
-		;;
-	esac
 fi
 
 AM_CONDITIONAL(HAVE_GCC, test "x$GCC" = "xyes")
--- harfbuzz/src/hb-private.hh	2016-07-01 10:33:10.072811100 +0200
+++ harfbuzz/src/hb-private.hh.clang	2016-07-01 10:33:52.207363000 +0200
@@ -83,7 +83,7 @@ extern "C" void  hb_free_impl(void *ptr)
 #define unlikely(expr) (expr)
 #endif
 
-#ifndef __GNUC__
+#if !defined(__GNUC__) && !defined(__clang__)
 #undef __attribute__
 #define __attribute__(x)
 #endif
