--- gnutls/gl/m4/socklen.m4	2017-10-09 11:30:24.468762000 +0200
+++ gnutls/gl/m4/socklen.m4.msvc81	2017-10-09 11:23:02.699446000 +0200
@@ -58,6 +58,8 @@ AC_DEFUN([gl_SOCKET_HEADERS],
 #if HAVE_SYS_SOCKET_H
 # include <sys/socket.h>
 #elif HAVE_WS2TCPIP_H
+# undef WINAPI_FAMILY
+# define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 # include <ws2tcpip.h>
 #endif
 ])
--- gnutls/gl/m4/sockpfaf.m4	2016-05-28 10:02:45.000000000 +0200
+++ gnutls/gl/m4/sockpfaf.m4.msvc81	2017-10-09 11:30:34.516662500 +0200
@@ -46,6 +46,8 @@ AC_DEFUN([gl_SOCKET_FAMILIES],
 #include <netinet/in.h>
 #endif
 #ifdef HAVE_WINSOCK2_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <winsock2.h>
 #endif
 #ifdef HAVE_WS2TCPIP_H
@@ -75,6 +77,8 @@ AC_DEFUN([gl_SOCKET_FAMILY_UNIX],
 #include <sys/un.h>
 #endif
 #ifdef HAVE_WINSOCK2_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <winsock2.h>
 #endif]],
 [[int x = AF_UNIX; struct sockaddr_un y;
--- gnutls/gl/m4/sys_socket_h.m4	2016-05-28 10:02:45.000000000 +0200
+++ gnutls/gl/m4/sys_socket_h.m4.msvc81	2017-10-09 12:03:25.547631400 +0200
@@ -59,6 +59,8 @@ AC_DEFUN([gl_HEADER_SYS_SOCKET],
 #include <sys/socket.h>
 #endif
 #ifdef HAVE_WS2TCPIP_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <ws2tcpip.h>
 #endif
 ])
@@ -77,6 +79,8 @@ AC_DEFUN([gl_HEADER_SYS_SOCKET],
        #include <sys/socket.h>
        #endif
        #ifdef HAVE_WS2TCPIP_H
+       #undef WINAPI_FAMILY
+       #define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
        #include <ws2tcpip.h>
        #endif
       ])
--- gnutls/gl/m4/sys_time_h.m4	2016-05-28 10:02:45.000000000 +0200
+++ gnutls/gl/m4/sys_time_h.m4.msvc81	2017-10-09 11:24:33.307049100 +0200
@@ -38,6 +38,8 @@ AC_DEFUN([gl_HEADER_SYS_TIME_H_BODY],
             #endif
             #include <time.h>
             #if HAVE_WINSOCK2_H
+            # undef WINAPI_FAMILY
+            # define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
             # include <winsock2.h>
             #endif
           ]],
@@ -63,6 +65,8 @@ AC_DEFUN([gl_HEADER_SYS_TIME_H_BODY],
               #endif
               #include <time.h>
               #if HAVE_WINSOCK2_H
+              # undef WINAPI_FAMILY
+              # define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
               # include <winsock2.h>
               #endif
             ]],
--- gnutls/gl/sys_socket.in.h	2016-05-28 10:02:45.000000000 +0200
+++ gnutls/gl/sys_socket.in.h.msvc81	2017-10-09 11:55:56.072728200 +0200
@@ -160,6 +160,8 @@ struct sockaddr_storage
    releases. */
 
 # if @HAVE_WINSOCK2_H@
+#  undef WINAPI_FAMILY
+#  define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #  include <winsock2.h>
 # endif
 # if @HAVE_WS2TCPIP_H@
--- gnutls/lib/gnutls_int.h	2015-12-15 22:29:11.000000000 +0100
+++ gnutls/lib/gnutls_int.h.msvc81	2017-10-09 13:45:45.001940800 +0200
@@ -54,6 +54,8 @@ typedef int ssize_t;
 #if HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #elif HAVE_WS2TCPIP_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <ws2tcpip.h>
 #endif
 #include <time.h>
--- gnutls/lib/gnutls_x509.c	2017-07-12 09:52:10.588755500 +0200
+++ gnutls/lib/gnutls_x509.c.msvc81	2017-10-09 13:03:43.073793700 +0200
@@ -48,6 +48,8 @@
 #include "system-keys.h"
 #include "urls.h"
 #ifdef _WIN32
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <wincrypt.h>
 #endif
 
--- gnutls/lib/inet_pton.c	2017-10-09 14:13:13.478643600 +0200
+++ gnutls/lib/inet_pton.c.msvc81	2017-10-09 14:03:11.570664900 +0200
@@ -30,6 +30,8 @@
 #endif
 
 #ifdef HAVE_WINSOCK2_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <winsock2.h>	/* needed to define AF_ values on Windows */
 #if _MSC_VER < 1600	/* errno.h defines EAFNOSUPPORT in Windows VC10 (and presumably eventually in VC11 ...) */
 #define EAFNOSUPPORT    WSAEAFNOSUPPORT
--- gnutls/lib/nettle/rnd-common.c	2015-01-19 18:38:09.000000000 +0100
+++ gnutls/lib/nettle/rnd-common.c.msvc81	2017-10-09 13:20:15.025872800 +0200
@@ -27,6 +27,10 @@
  * and modified to fit gnutls' needs. Relicenced with permission. 
  * Original author Niels Möller.
  */
+ 
+ #ifdef _WIN32
+#include <wincrypt.h>
+#endif
 
 #include <gnutls_int.h>
 #include <gnutls_errors.h>
--- gnutls/lib/system.c	2017-07-12 09:52:10.349255600 +0200
+++ gnutls/lib/system.c.msvc81	2017-10-09 13:26:12.825712300 +0200
@@ -20,6 +20,10 @@
  *
  */
 
+#ifdef _WIN32
+#include <wincrypt.h>
+#endif
+
 #include <config.h>
 #include <system.h>
 #include <gnutls_int.h>
@@ -32,8 +36,8 @@
 #include <c-ctype.h>
 
 #ifdef _WIN32
-# include <windows.h>
-# include <wincrypt.h>
+# undef WINAPI_FAMILY
+# define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 # include <winapifamily.h>
 # if (WINAPI_FAMILY == WINAPI_FAMILY_PC_APP || WINAPI_FAMILY == WINAPI_FAMILY_PHONE_APP)
 #  if defined(_WIN32_WINNT) && _WIN32_WINNT >= 0x0A00 /* Univeral Winstore */
--- gnutls/lib/system.h	2015-07-21 11:42:08.000000000 +0200
+++ gnutls/lib/system.h.msvc81	2017-10-09 19:09:15.110024300 +0200
@@ -33,6 +33,11 @@
 #include <windows.h>		/* for Sleep */
 #endif
 
+#  undef WINAPI_FAMILY
+#  define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
+#  include <winsock2.h>
+
+
 #ifdef _POSIX_PATH_MAX
 # define GNUTLS_PATH_MAX _POSIX_PATH_MAX
 #else
--- gnutls/lib/x509/hostname-verify.c	2015-03-28 13:06:31.000000000 +0100
+++ gnutls/lib/x509/hostname-verify.c.msvc81	2017-10-09 12:48:24.881456300 +0200
@@ -27,6 +27,13 @@
 #include <system.h>
 #include <gnutls-idna.h>
 
+#if defined(_MSC_VER) && _MSC_VER < 1900
+# undef WINAPI_FAMILY
+# define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
+# include <ws2def.h>
+# include <ws2ipdef.h>
+#endif
+
 /**
  * gnutls_x509_crt_check_hostname:
  * @cert: should contain an gnutls_x509_crt_t type
--- gnutls/lib/x509/output.c	2017-07-12 09:52:10.724254700 +0200
+++ gnutls/lib/x509/output.c.msvc81	2017-10-09 12:44:13.585184900 +0200
@@ -36,6 +36,11 @@
 #if defined(HAVE_INET_NTOP) && !defined(_MSC_VER)
 # include <arpa/inet.h>
 #endif
+#if defined(_MSC_VER) && _MSC_VER < 1900
+# undef WINAPI_FAMILY
+# define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
+# include <ws2def.h>
+#endif
 
 #define addf _gnutls_buffer_append_printf
 #define adds _gnutls_buffer_append_str
--- gnutls/src/certtool-cfg.c	2016-02-06 20:50:36.000000000 +0100
+++ gnutls/src/certtool-cfg.c.msvc81	2017-10-09 13:46:35.075968300 +0200
@@ -45,6 +45,8 @@
 #include <sys/socket.h>
 #include <arpa/inet.h>
 #elif HAVE_WS2TCPIP_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <ws2tcpip.h>
 #endif
 
--- gnutls/src/cli-debug.c	2016-04-27 07:39:27.000000000 +0200
+++ gnutls/src/cli-debug.c.msvc81	2017-10-09 13:46:43.682079800 +0200
@@ -29,6 +29,8 @@
 #if HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #elif HAVE_WS2TCPIP_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <ws2tcpip.h>
 #endif
 #include <tests.h>
--- gnutls/src/cli.c	2016-05-19 18:27:48.000000000 +0200
+++ gnutls/src/cli.c.msvc81	2017-10-09 13:46:45.793794500 +0200
@@ -31,6 +31,8 @@
 #if HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #elif HAVE_WS2TCPIP_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <ws2tcpip.h>
 #endif
 #include <sys/select.h>
--- gnutls/src/gl/m4/getaddrinfo.m4	2015-01-19 18:38:09.000000000 +0100
+++ gnutls/src/gl/m4/getaddrinfo.m4.msvc81	2017-10-09 13:47:42.069180600 +0200
@@ -46,6 +46,8 @@ AC_DEFUN([gl_GETADDRINFO],
       LIBS="$LIBS -lws2_32"
       AC_LINK_IFELSE([AC_LANG_PROGRAM([[
 #ifdef HAVE_WS2TCPIP_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <ws2tcpip.h>
 #endif
 #include <stddef.h>
@@ -72,6 +74,8 @@ AC_DEFUN([gl_GETADDRINFO],
 #include <netdb.h>
 #endif
 #ifdef HAVE_WS2TCPIP_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <ws2tcpip.h>
 #endif
 #include <stddef.h>
@@ -86,6 +90,8 @@ AC_DEFUN([gl_GETADDRINFO],
 #include <netdb.h>
 #endif
 #ifdef HAVE_WS2TCPIP_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <ws2tcpip.h>
 #endif
 #include <stddef.h>
@@ -102,6 +108,8 @@ AC_DEFUN([gl_GETADDRINFO],
 #include <netdb.h>
 #endif
 #ifdef HAVE_WS2TCPIP_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <ws2tcpip.h>
 #endif
 #include <stddef.h>
@@ -157,6 +165,8 @@ AC_DEFUN([gl_PREREQ_GETADDRINFO], [
 #include <netdb.h>
 #endif
 #ifdef HAVE_WS2TCPIP_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <ws2tcpip.h>
 #endif
 ]])
@@ -182,6 +192,8 @@ AC_DEFUN([gl_PREREQ_GETADDRINFO], [
 #include <netdb.h>
 #endif
 #ifdef HAVE_WS2TCPIP_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <ws2tcpip.h>
 #endif
 ])
--- gnutls/src/gl/m4/hostent.m4	2015-01-19 18:38:09.000000000 +0100
+++ gnutls/src/gl/m4/hostent.m4.msvc81	2017-10-09 11:24:49.850437400 +0200
@@ -32,6 +32,8 @@ AC_DEFUN([gl_HOSTENT],
            [AC_LANG_PROGRAM(
               [[
 #ifdef HAVE_WINSOCK2_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <winsock2.h>
 #endif
 #include <stddef.h>
--- gnutls/src/gl/m4/inet_ntop.m4	2015-01-19 18:38:09.000000000 +0100
+++ gnutls/src/gl/m4/inet_ntop.m4.msvc81	2017-10-09 13:48:00.829070200 +0200
@@ -23,7 +23,10 @@ AC_DEFUN([gl_FUNC_INET_NTOP],
   INET_NTOP_LIB=
   gl_PREREQ_SYS_H_WINSOCK2
   if test $HAVE_WINSOCK2_H = 1; then
-    AC_CHECK_DECLS([inet_ntop],,, [[#include <ws2tcpip.h>]])
+    AC_CHECK_DECLS([inet_ntop],,, [[
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
+#include <ws2tcpip.h>]])
     if test $ac_cv_have_decl_inet_ntop = yes; then
       dnl It needs to be overridden, because the stdcall calling convention
       dnl is not compliant with POSIX.
--- gnutls/src/gl/m4/inet_pton.m4	2015-01-19 18:38:09.000000000 +0100
+++ gnutls/src/gl/m4/inet_pton.m4.msvc81	2017-10-09 13:48:13.380814300 +0200
@@ -23,7 +23,10 @@ AC_DEFUN([gl_FUNC_INET_PTON],
   INET_PTON_LIB=
   gl_PREREQ_SYS_H_WINSOCK2
   if test $HAVE_WINSOCK2_H = 1; then
-    AC_CHECK_DECLS([inet_pton],,, [[#include <ws2tcpip.h>]])
+    AC_CHECK_DECLS([inet_pton],,, [[
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
+#include <ws2tcpip.h>]])
     if test $ac_cv_have_decl_inet_pton = yes; then
       dnl It needs to be overridden, because the stdcall calling convention
       dnl is not compliant with POSIX.
--- gnutls/src/gl/m4/servent.m4	2015-01-19 18:38:09.000000000 +0100
+++ gnutls/src/gl/m4/servent.m4.msvc81	2017-10-09 11:24:56.338499500 +0200
@@ -34,6 +34,8 @@ AC_DEFUN([gl_SERVENT],
            [AC_LANG_PROGRAM(
               [[
 #ifdef HAVE_WINSOCK2_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <winsock2.h>
 #endif
 #include <stddef.h>
--- gnutls/src/gl/m4/socketlib.m4	2015-01-19 18:38:09.000000000 +0100
+++ gnutls/src/gl/m4/socketlib.m4.msvc81	2017-10-09 11:25:03.050711600 +0200
@@ -20,6 +20,8 @@ AC_DEFUN([gl_SOCKETLIB],
       LIBS="$LIBS -lws2_32"
       AC_LINK_IFELSE([AC_LANG_PROGRAM([[
 #ifdef HAVE_WINSOCK2_H
+# undef WINAPI_FAMILY
+# define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 # include <winsock2.h>
 #endif]], [[
           WORD wVersionRequested = MAKEWORD(1, 1);
--- gnutls/src/gl/sys_socket.in.h	2015-01-19 18:38:09.000000000 +0100
+++ gnutls/src/gl/sys_socket.in.h.msvc81	2017-10-09 12:37:51.112976300 +0200
@@ -160,6 +160,8 @@ struct sockaddr_storage
    releases. */
 
 # if @HAVE_WINSOCK2_H@
+#  undef WINAPI_FAMILY
+#  define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #  include <winsock2.h>
 # endif
 # if @HAVE_WS2TCPIP_H@
--- gnutls/src/libopts/compat/windows-config.h	2015-11-28 09:20:33.000000000 +0100
+++ gnutls/src/libopts/compat/windows-config.h.msvc81	2017-10-09 11:25:32.215362100 +0200
@@ -76,6 +76,8 @@
 
 /* Include Windows headers */
 #include <windows.h>
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <winsock2.h>
 #include <limits.h>
 
--- gnutls/src/socket.c	2016-05-19 18:27:48.000000000 +0200
+++ gnutls/src/socket.c.msvc81	2017-10-09 13:46:48.970193400 +0200
@@ -22,6 +22,8 @@
 #if HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #elif HAVE_WS2TCPIP_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <ws2tcpip.h>
 #endif
 #include <netdb.h>
--- gnutls/src/udp-serv.c	2015-01-19 18:38:09.000000000 +0100
+++ gnutls/src/udp-serv.c.msvc81	2017-10-09 13:46:54.217853500 +0200
@@ -23,6 +23,8 @@
 #if HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #elif HAVE_WS2TCPIP_H
+#undef WINAPI_FAMILY
+#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
 #include <ws2tcpip.h>
 #endif
 #include <arpa/inet.h>
