--- ffmpeg/compat/w32pthreads.h	2016-06-15 09:52:13.000000000 +0200
+++ ffmpeg/compat/w32pthreads.h.clang	2016-07-05 14:51:52.473780400 +0200
@@ -88,7 +88,7 @@ static av_unused int pthread_create(pthr
     thread->func   = start_routine;
     thread->arg    = arg;
 #if HAVE_WINRT
-    thread->handle = (void*)CreateThread(NULL, 0, win32thread_worker, thread,
+    thread->handle = (void*)CreateThread(NULL, 0, (LPTHREAD_START_ROUTINE) win32thread_worker, thread,
                                            0, NULL);
 #else
     thread->handle = (void*)_beginthreadex(NULL, 0, win32thread_worker, thread,
--- ffmpeg/libavutil/attributes.h	2016-06-15 09:52:28.000000000 +0200
+++ ffmpeg/libavutil/attributes.h.clang	2016-07-05 14:57:20.008979500 +0200
@@ -122,7 +122,7 @@
 #endif
 
 
-#if defined(__GNUC__)
+#if defined(__GNUC__) || defined(__clang__)
 #    define av_unused __attribute__((unused))
 #else
 #    define av_unused
--- ffmpeg/libavcodec/neon/mpegvideo.c	2016-06-15 09:52:19.000000000 +0200
+++ ffmpeg/libavcodec/neon/mpegvideo.c.clang	2016-07-05 16:53:22.554366700 +0200
@@ -125,10 +125,12 @@ static void dct_unquantize_h263_intra_ne
 
 av_cold void ff_mpv_common_init_neon(MpegEncContext *s)
 {
+#if !defined(_MSC_VER) || !defined(__clang__)
     int cpu_flags = av_get_cpu_flags();
 
     if (have_neon(cpu_flags)) {
         s->dct_unquantize_h263_intra = dct_unquantize_h263_intra_neon;
         s->dct_unquantize_h263_inter = dct_unquantize_h263_inter_neon;
     }
+#endif
 }
