--- ffmpeg-HEAD/configure	2016-06-08 18:40:07.000000000 +0200
+++ ffmpeg-HEAD/configure.msvc	2016-06-08 18:43:53.302017200 +0200
@@ -2806,6 +2806,8 @@ case "$toolchain" in
         ld_default="$source_path/compat/windows/mslink"
         nm_default="dumpbin -symbols"
         ar_default="lib"
+        LIBPREF=""
+        LIBSUF=".lib"
         case "$arch" in
         arm*)
             as_default="armasm"
@@ -2986,12 +2988,12 @@ msvc_common_flags(){
             -fPIC)                ;;
             -mthumb)              ;;
             -march=*)             ;;
-            -lz)                  echo zlib.lib ;;
+            -lz)                  echo z.lib ;;
             -lavifil32)           echo vfw32.lib ;;
             -lavicap32)           echo vfw32.lib user32.lib ;;
             -lx264)               echo libx264.lib ;;
             -l*)                  echo ${flag#-l}.lib ;;
-            -L*)                  echo -libpath:${flag#-L} ;;
+            -L*)                  echo -libpath:`cygpath -w ${flag#-L}` ;;
             *)                    echo $flag ;;
         esac
     done
@@ -3330,6 +3332,10 @@ host_ldflags_filter=$_flags_filter
 add_host_ldflags $_flags $_ldflags
 HOSTLD_O=${_ld_o-$HOSTLD_O}
 
+_ldflags=$LDFLAGS
+LDFLAGS=
+add_ldflags $_ldflags
+
 if [ -z "$CC_DEPFLAGS" ] && [ "$dep_cc" != "$cc" ]; then
     probe_cc depcc "$dep_cc"
     CCDEP=${_DEPCMD:-$DEPCMD}
