--- ffmpeg/configure    2015-08-18 13:35:16.617998500 +0200
+++ ffmpeg/configure.msvc       2015-08-18 13:31:20.171290900 +0200
@@ -2642,6 +2642,8 @@
         ld_default="$source_path/compat/windows/mslink"
         nm_default="dumpbin -symbols"
         ar_default="lib"
+        LIBPREF=""
+        LIBSUF=".lib"
         case "$arch" in
         arm*)
             as_default="armasm"
@@ -2817,11 +2819,11 @@
             -fPIC)                ;;
             -mthumb)              ;;
             -march=*)             ;;
-            -lz)                  echo zlib.lib ;;
+            -lz)                  echo z.lib ;;
             -lavifil32)           echo vfw32.lib ;;
             -lavicap32)           echo vfw32.lib user32.lib ;;
             -l*)                  echo ${flag#-l}.lib ;;
-            -L*)                  echo -libpath:${flag#-L} ;;
+            -L*)                  echo -libpath:`cygpath -w ${flag#-L}` ;;
             *)                    echo $flag ;;
         esac
     done
@@ -3154,6 +3156,10 @@
 add_host_ldflags $_flags $_ldflags
 HOSTLD_O=${_ld_o-$HOSTLD_O}

+_ldflags=$LDFLAGS
+LDFLAGS=
+add_ldflags $_ldflags
+
 if [ -z "$CC_DEPFLAGS" ] && [ "$dep_cc" != "$cc" ]; then
     probe_cc depcc "$dep_cc"
     CCDEP=${_DEPCMD:-$DEPCMD}

