diff --git a/configure b/configure
index b10715f..3d26996 100755
--- a/configure
+++ b/configure
@@ -2821,7 +2821,7 @@
             -lavifil32)           echo vfw32.lib ;;
             -lavicap32)           echo vfw32.lib user32.lib ;;
             -l*)                  echo ${flag#-l}.lib ;;
-            -L*)                  echo -libpath:${flag#-L} ;;
+            -L*)                  echo -libpath:`cygpath -w ${flag#-L}` ;;
             *)                    echo $flag ;;
         esac
     done
@@ -3154,6 +3154,10 @@
 add_host_ldflags $_flags $_ldflags
 HOSTLD_O=${_ld_o-$HOSTLD_O}
 
+_ldflags=$LDFLAGS
+LDFLAGS=
+add_ldflags $_ldflags
+
 if [ -z "$CC_DEPFLAGS" ] && [ "$dep_cc" != "$cc" ]; then
     probe_cc depcc "$dep_cc"
     CCDEP=${_DEPCMD:-$DEPCMD}
--- ffmpeg/configure    2015-08-17 09:18:14.227153000 +0200
+++ ffmpeg/configure.msvc       2015-08-17 09:20:58.307096800 +0200
@@ -2642,6 +2642,8 @@
         ld_default="$source_path/compat/windows/mslink"
         nm_default="dumpbin -symbols"
         ar_default="lib"
+        LIBPREF=""
+        LIBSUF=".lib"
         case "$arch" in
         arm*)
             as_default="armasm"

