--- ffmpeg/configure.old	2016-02-04 07:50:15.006599300 +0100
+++ ffmpeg/configure	2016-02-19 11:18:02.706616300 +0100
@@ -2697,6 +2697,7 @@ case "$toolchain" in
             as_default="armasm"
             ;;
         esac
+        add_extralibs mincore.lib
         target_os_default="win32"
         # Use a relative path for TMPDIR. This makes sure all the
         # ffconf temp files are written with a relative path, avoiding
