--- aribb24/src/convtable.h     2014-08-18 17:29:48.000000000 +0200
+++ aribb24/src/convtable.h.msvc        2015-08-12 17:19:32.222469300 +0200
@@ -165,19 +165,19 @@
         0x2517, 0x2523, 0x2533, 0x252b, 0x253b, 0x254b, 0x2520, 0x252f,
         0x2528, 0x2537, 0x253f, 0x251d, 0x2530, 0x2525, 0x2538, 0x2542,
     },
-    {
+    {0x0000
     },
-    {
+    {0x0000
     },
-    {
+    {0x0000
     },
-    {
+    {0x0000
     },
-    {
+    {0x0000
     },
-    {
+    {0x0000
     },
-    {
+    {0x0000
     },
     {
         0x4e9c, 0x5516, 0x5a03, 0x963f, 0x54c0, 0x611b, 0x6328, 0x59f6,
@@ -1151,11 +1151,11 @@
         0x9321, 0x9348, 0x9592, 0x96de, 0x9903, 0x9940, 0x9ad9, 0x9bd6,
         0x9dd7, 0x9eb4, 0x9eb5,
     },
-    {
+    {0x0000
     },
-    {
+    {0x0000
     },
-    {
+    {0x0000
     },
     {
         0xe0c9, 0xe0ca, 0xe0cb, 0xe0cc, 0xe0cd, 0xe0ce, 0x0000, 0xe0d0,

--- aribb24/src/md5.c   2014-08-18 17:29:48.000000000 +0200
+++ aribb24/src/md5.c.msvc      2015-08-12 17:19:32.227472400 +0200
@@ -334,7 +334,7 @@
     md5_init( h );
 }

-void AddMD5( struct md5_s *restrict h, const void *data, size_t len )
+void AddMD5( struct md5_s * h, const void *data, size_t len )
 {
     md5_write( h, data, len );
 }

