--- game-music-emu-0.6.0/gme/CMakeLists.txt.orig	2012-05-27 22:48:53.000000000 +0200
+++ game-music-emu-0.6.0/gme/CMakeLists.txt	2013-07-04 19:48:23.031372405 +0200
@@ -143,7 +143,7 @@
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 # Add library to be compiled.
-add_library(gme SHARED ${libgme_SRCS})
+add_library(gme STATIC ${libgme_SRCS})
 
 # The version is the release.  The "soversion" is the API version.  As long
 # as only build fixes are performed (i.e. no backwards-incompatible changes
--- game-music-emu/gme/CMakeLists.txt	2016-12-16 14:29:02.726746500 +0100
+++ game-music-emu/gme/CMakeLists.txt.win32	2016-12-16 14:27:30.866603200 +0100
@@ -137,7 +137,7 @@ configure_file(${CMAKE_CURRENT_SOURCE_DI
 # On some platforms we may need to change headers or whatnot based on whether
 # we're building the library or merely using the library. The following is
 # only defined when building the library to allow us to tell which is which.
-add_definitions(-DBLARGG_BUILD_DLL)
+#add_definitions(-DBLARGG_BUILD_DLL)
 
 # For the gme_types.h
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
