--- libvpx/configure	2017-01-12 21:27:27.000000000 +0100
+++ libvpx/configure.msvc	2017-04-14 15:37:48.968931200 +0200
@@ -106,6 +106,7 @@ all_platforms="${all_platforms} armv7-da
 all_platforms="${all_platforms} armv7-linux-rvct"    #neon Cortex-A8
 all_platforms="${all_platforms} armv7-linux-gcc"     #neon Cortex-A8
 all_platforms="${all_platforms} armv7-none-rvct"     #neon Cortex-A8
+all_platforms="${all_platforms} armv7-win32-vsgcc"   #neon Cortex-A8
 all_platforms="${all_platforms} armv7-win32-vs11"
 all_platforms="${all_platforms} armv7-win32-vs12"
 all_platforms="${all_platforms} armv7-win32-vs14"
@@ -131,6 +132,7 @@ all_platforms="${all_platforms} x86-linu
 all_platforms="${all_platforms} x86-os2-gcc"
 all_platforms="${all_platforms} x86-solaris-gcc"
 all_platforms="${all_platforms} x86-win32-gcc"
+all_platforms="${all_platforms} x86-win32-vsgcc"
 all_platforms="${all_platforms} x86-win32-vs10"
 all_platforms="${all_platforms} x86-win32-vs11"
 all_platforms="${all_platforms} x86-win32-vs12"
@@ -148,6 +150,7 @@ all_platforms="${all_platforms} x86_64-l
 all_platforms="${all_platforms} x86_64-linux-icc"
 all_platforms="${all_platforms} x86_64-solaris-gcc"
 all_platforms="${all_platforms} x86_64-win64-gcc"
+all_platforms="${all_platforms} x86_64-win64-vsgcc"
 all_platforms="${all_platforms} x86_64-win64-vs10"
 all_platforms="${all_platforms} x86_64-win64-vs11"
 all_platforms="${all_platforms} x86_64-win64-vs12"
@@ -637,6 +640,9 @@ process_toolchain() {
 
     # Enable the postbuild target if building for visual studio.
     case "$tgt_cc" in
+        vsgcc) enable_feature msvs
+             INLINE="__forceinline"
+        ;;
         vs*) enable_feature msvs
              enable_feature solution
              vs_version=${tgt_cc##vs}
--- libvpx/build/make/configure.sh	2017-04-14 15:39:39.738893800 +0200
+++ libvpx/build/make/configure.sh.msvc	2017-04-14 15:38:51.929300400 +0200
@@ -1121,6 +1121,12 @@ EOF
           #for 32 bit x86 builds, -O3 did not turn on this flag
           enabled optimizations && disabled gprof && check_add_cflags -fomit-frame-pointer
           ;;
+        vsgcc)
+          AS=auto
+          msvs_arch_dir=x86-msvs
+          #for 32 bit x86 builds, -O3 did not turn on this flag
+          enabled optimizations && disabled gprof && check_add_cflags -fomit-frame-pointer
+          ;;
         vs*)
           # When building with Microsoft Visual Studio the assembler is
           # invoked directly. Checking at configure time is unnecessary.
